runtime/common/RuntimeMLIR.cpp

```diff
@@ -94,12 +94,20 @@ bool setupTargetTriple(llvm::Module *llvmModule) {
   return true;
 }
 
-void optimizeLLVM(llvm::Module *module, bool enableOpt) {
+void optimizeLLVM(llvm::Module *module) {
+  bool enableOpt = true;
   auto optPipeline = makeOptimizingTransformer(
       /*optLevel=*/enableOpt ? 3 : 0, /*sizeLevel=*/0,
       /*targetMachine=*/nullptr);
```

Comment:
May I ask the reason behind converting `enableOpt` to a local variable? I'd think having it as an argument with default value (`true`) is more flexible. Otherwise, if we don't allow disabling optimization in context, we can just set the `optLevel` to 3 in the `makeOptimizingTransformer` call.

Reply:
`makeOptimizingTransformer()` is an LLVM function, so we cannot change out it operates.

Also, we cannot disable optimization right now. The `remove-measurements` pass removes the following calls (as generated from a stripped down version of `load_values.cpp` where I just generated 1 qubit:

```diff
define void @__nvqpp__mlirgen__function_load_value._Z10load_valuej() #1 {
  %1 = alloca [1 x i1], align 1
-  call void @__quantum__qis__mz__body(%Qubit* null, %Result* writeonly null)
  %2 = load i1, i1* null, align 1
  %3 = bitcast [1 x i1]* %1 to i1*
  store i1 %2, i1* %3, align 1
-  call void @__quantum__rt__result_record_output(%Result* null, i8* getelementptr inbounds ([7 x i8], [7 x i8]* @cstr.72303030303000, i32 0, i32 0))
  ret void
}
```
But that leaves `alloca`, `load`, `bitcast`, and `store` operations in the program, which are not valid Base Profile instructions, and I am currently counting on optimization to remove them. I tried to simply remove them with the `cse` pass, but that did not work. I'm open to other suggestions if you have them.

Note that as written, this change reverts this section of the code back to how it was prior to #656, so it's at least similar to how it was before.

Reply:
Thanks for the explanation. What I meant in the comment is that we can simplify the above code in this case (disallow turning off optimization) to
```
void optimizeLLVM(llvm::Module *module) {
  auto optPipeline = makeOptimizingTransformer(
      /*optLevel=*/3, /*sizeLevel=*/0,
      /*targetMachine=*/nullptr);
```      

(removing the local variable `enableOpt = true`)

Reply:
Updated in 7926e58b.